### PR TITLE
fix: added healthcheck for editor with https

### PR DESCRIPTION
**Pull request by sum35h**
## Description
Added a fix for the docker health check on editor with ssl enabled for `appsmith-ce` fat container and the slim client `appsmith-editor`

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: FIX/health-check-https 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

**Review by pratapaprasanna — DISMISSED** [review]

File: app/client/Dockerfile [hunk 1]
```diff
@@ -16,5 +17,5 @@ COPY ./docker/templates/nginx-app-https.conf.template /nginx-app-https.conf.temp
 
 # This is the script that is used to start Nginx when the Docker container starts
 COPY ./docker/start-nginx.sh /start-nginx.sh
-HEALTHCHECK --interval=15s --timeout=15s --start-period=15s --retries=3 CMD curl -f http://localhost:80/ || exit 1
+HEALTHCHECK --interval=15s --timeout=15s --start-period=15s --retries=3 CMD curl -f http://localhost/ || curl -fk https://localhost/ || exit 1
```
Comment (sharat87) [inline app/client/Dockerfile:20]: Do we need to check for both http and https? The http URL will redirect to HTTPS when available right? So, just adding `-L` to the http curl command will address this I think?

Comment (sum35h) [reply app/client/Dockerfile:20]: Yes, that looks good. Wasn't aware of that option.

File: deploy/docker/scripts/healthcheck.sh [hunk 2]
```diff
@@ -11,9 +11,11 @@ while read -r line
     else
       echo "PROCESS: $process - STATUS: $status"
       if [[ "$process" == 'editor' ]]; then
-        if [[ $(curl -s -w "%{http_code}\n" http://localhost:80/ -o /dev/null) -ne 200 ]]; then
+        if [[ $(curl -s -w "%{http_code}\n" http://localhost/ -o /dev/null) -ne 200 ]]; then
```
Comment (sharat87) [inline deploy/docker/scripts/healthcheck.sh:14]: Could we do the same thing here as well please? Using `-L`?

**Review by sharat87 — CHANGES_REQUESTED** [review]

File: deploy/docker/scripts/healthcheck.sh (hunk 2 above)
Comment (sum35h) [reply deploy/docker/scripts/healthcheck.sh:14]: @sharat87 I checked it with -L here, but the status code still comes to be 301.

**Review by sharat87 — APPROVED** [review]

**Review by pratapaprasanna — APPROVED** [review]

**Pull request merged**